### DISCUSSION TITLE
【求助】在编辑器行号旁创建图形ICON

### DISCUSSION THREAD

**Discussion by MilkWangStudio** [opensumi/core#3012] [Q&A] [answered]
类似这种效果，在sql行号上标注执行中、执行成功、失败的icon
![image](https://github.com/opensumi/core/assets/15866249/278e3258-742d-427b-808d-cda36ba70281)

monaco中时使用createDecorationsCollection这个API，请问在opensumi中有什么api可以做到吗。

**Comment by Ricbet** [comment, marked as answer]
在 opensumi 中可以使用 monaco 的 createDecorationsCollection 这个 API

```typescript
// IEditor
editor.monacoEditor.createDecorationsCollection()
```
